core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp

```diff
@@ -670,7 +677,7 @@ class ParallelScan<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::Cuda> {
   }
 
   ParallelScan(const FunctorType& arg_functor, const Policy& arg_policy)
-      : m_functor(arg_functor),
+      : m_functor_reducer(arg_functor, typename Analysis::Reducer{arg_functor}),
```

Comment:
This (and another PR?) are the changes that prevent the FunctorAnalysis object from being created on the device now?

Reply:
Indirectly. The point really is to store the `FunctorAnaysis::Reducer` object (wrapping the functor) instead of storing the functor directly (and thus creating a copy on the device when creating the `FunctorAnalysis::Reducer` object).